File: learning-akka/src/main/java/book/java/ch1/AkkademyDb.java
package book.java.ch1;

import akka.actor.AbstractActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;
import akka.japi.pf.ReceiveBuilder;
import book.kotlin.messages.SetRequest;

import java.util.HashMap;
import java.util.Map;

public class AkkademyDb extends AbstractActor {

	final LoggingAdapter log = Logging.getLogger(context().system(), this);
	final Map<String, Object> map = new HashMap<>();

	private AkkademyDb() {
		receive(ReceiveBuilder.
				match(SetRequest.class, message -> {
					log.info("Received set request – key: {}; value: {}", message.getKey(), message.getValue());
					map.put(message.getKey(), message.getValue());
				}).
				matchAny(o -> log.info("received unknown message: {}", o))
				.build()
		);
	}

}

File: learning-akka/src/main/kotlin/book/kotlin/ch1/AkkademyDb.kt
package book.kotlin.ch1

import akka.actor.UntypedActor
import akka.event.Logging
import book.kotlin.messages.SetRequest
import book.kotlin.util.system

class AkkademyDb : UntypedActor() {

    val log = Logging.getLogger(context.system, this)
    val map = mutableMapOf<String, Any>()

    override fun onReceive(message: Any) = when (message) {
        is SetRequest -> {
            log.info("received SetRequest - key: {}; value: {}", message.key, message.value)
            map += message.key to message.value
        }
        else -> log.info("received unknown message: {}", message)
    }

}